### PR TITLE
Web and pdflib enhancements

### PR DESCRIPTION
**Pull request by dcbr**
Awesome library! I would like to use it for web applications, in combination with the pdf-lib library, for which following enhancements are required or useful.

## Overview

- [Web]: As there is no `node:Buffer` type, web applications are not supported right now (type error). This PR accepts beside the `Buffer` also the `Uint8Array` (both available in Node and Web applications) and `string` (base64 encoded bytes) types when signing a pdf or creating a `P12Signer`. An extra utility function `convertBuffer` is introduced for this.
- [Pdflib]: The pdflib placeholder now accepts a `PDFDocument` or `PDFPage` input, supporting signatures on other pages than the first one.
- [Pdflib]: The pdflib placeholder now accepts a `signingTime` input, such that it can be matched with the document creation or modification time.
- [Pdflib]: The pdflib placeholder inserts the signature dictionary as a `PDFInvalidObject`, such that it is not serialized in an object stream. This means the `{useObjectStreams: false}` is no longer needed when saving the document (and thus compressing the final signed pdf size).
- [Pdflib]: The pdflib placeholder inserts an appearance stream dictionary for the signature widget, which is required for PDF/A compliance.
- [Pdflib]: The pdflib placeholder inserts a `Prop_Build` dictionary for the signature. Allowing to specify which filter and application were used to generate the signature.
- [Sign]: The sign function and Signers have an extra `signingTime` input, such that it can be matched with the document creation or modification time.

### CODE REVIEW

**Comment by vbuch** [comment]
Reading through your description, this seems like an exciting contribution. I'll give it a more thorough look in the coming days but I love every word of the description.

File: packages/placeholder-pdf-lib/src/pdflibAddPlaceholder.js [hunk 1]
```diff
@@ -59,24 +71,37 @@ export const pdflibAddPlaceholder = ({
...
-        M: PDFString.fromDate(new Date()),
+        M: PDFString.fromDate(signingTime ?? new Date()),
         ContactInfo: PDFString.of(contactInfo),
         Name: PDFString.of(name),
         Location: PDFString.of(location),
-    }, pdfDoc.index);
-    const signatureDictRef = pdfDoc.context.register(signatureDict);
+        Prop_Build: {
+            Filter: {Name: 'Adobe.PPKLite'},
+            App: {Name: 'pdfsign'},
```
Comment (vbuch) [inline packages/placeholder-pdf-lib/src/pdflibAddPlaceholder.js:87]: We should allow the app to set its name that. We are just a helper library.

**Review by vbuch — APPROVED** [review]
All looks great and in line with what we've been doing so far. Thank you for a great contribution.

File: packages/placeholder-pdf-lib/src/pdflibAddPlaceholder.js (hunk 1 above)
Comment (dcbr) [reply packages/placeholder-pdf-lib/src/pdflibAddPlaceholder.js:87]: Alright, I will make it an extra parameter to the placeholder function then.
Do you squash all commits when merging or would you like me to rebase everything into 1 commit?

Comment (vbuch) [reply packages/placeholder-pdf-lib/src/pdflibAddPlaceholder.js]: I don't usually squash and I usually rebase and reword my commits a lot. But don't feel bothered by that. Whatever is fine for you. The working code is important.

**Review by vbuch — APPROVED** [review]

**Comment by vbuch** [comment]
I'll get that released today or tomorrow. Thank you for a great contribution!

**Pull request merged**